source/main.bmx

```diff
@@ -6159,6 +6159,11 @@ endrem
 			GetProgrammeProducerCollection().UpdateAll()
 		EndIf
 
+		'=== UPDATE AWARDS / SAMMYS ===
+		'when updating awards onHour, onDayBegins would fire before the last day's award was handled
+		'fail of mission possible although the award is won/wrong award count
+		If minute = 59 Then GetAwardCollection().UpdateAwards()
```

Comment:
Das bedeutet aber, wenn ich 23:59 beim Filmhaendler was verkaufe und somit ein "Geldziel" erreiche, wird das nun dafuer auch nicht mehr gewertet.

Ist unwahrscheinlich, aber .... war jetzt mehr "prinzipiell" gedacht.

Waere es nicht hmm... sinnstiftender, einen neuen GameEvent einzufuehren:
`Game_OnDayEnds`
und dann in game.game.bmx:
```
'hinzufuegen
			worldTime._timeGone :- 1

				TriggerBaseEvent(GameEventKeys.Game_OnDayEnds, New TData.AddLong("time", worldTime.GetTimeGone()))
			worldTime._timeGone :+ 1

				TriggerBaseEvent(GameEventKeys.Game_OnDay, New TData.AddLong("time", worldTime.GetTimeGone()))
```
Eventuell koennte man damit "das Fitzelchen an Zeit vor Mitternacht" zurueckgehen und auf das Tagesende reagieren.




Aber ich vermute, ich mache hier ein Fass auf, was gar nicht geoeffnet werden braucht ;) -- also gerne einfach ignorieren.

Reply:
Ja, ein OnDayEnds wäre die sauberste Lösung - auch für das Bezahlen der täglichen Sender- und Agenturgebühren.
Die Änderung auf 23:59 betrifft ja aber nur die Sammys. Ziel ist, dass der Sammy am Vortag vergeben wird, so dass er bei der OnDayBegins-Prüfung der Missionsziele schon da ist. Ansonsten würde er nämlich erst nach der Missionsprüfung vergeben, was spielentscheidend ist. Und Sammy-Punkte sollten ja eigentlich immer spätestens zur Minute 55 vergeben werden. Einzige Ausnahme sind Eigenproduktionen, die 23:59 enden, aber wenn ich mich richtig erinnere habe ich die Sammy-Vergabe ganz ans Ende von onMinute gepackt.

Alle anderen Missionen sollten überhaupt nicht betroffen sein.

Reply:
Eine Reihe anderer Aufrufe hier klingen auch eher nach OnDayEnds (Auktionen etc.). Das sollte aber in einem separaten Ticket behandelt werden, da die Auswirkungen intensiv geprüft werden sollten.